web-server/v0.4/src/components/SessionModal/index.js

```diff
@@ -0,0 +1,136 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { Modal, Form, Input, Button, message, Tooltip, Icon } from 'antd';
+import axios from 'axios';
+
+const { TextArea } = Input;
+
+export default class SessionModal extends Component {
+  static propTypes = {
+    configData: PropTypes.object.isRequired,
+    styles: PropTypes.object.isRequired,
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      description: '',
+      visible: false,
+      confirmLoading: false,
+      generatedLink: '',
+    };
+  }
+
+  showSuccess = () => {
+    const { generatedLink } = this.state;
+
+    Modal.success({
+      title: 'Generated session link',
+      content: (
+        <div style={{ display: 'flex', flex: 1, flexDirection: 'row' }}>
+          <Input id="generatedUrl" value={generatedLink} />
+          <Button style={{ marginLeft: 8 }} icon="copy" onClick={this.copyLink}>
+            Copy Link
+          </Button>
+        </div>
+      ),
+    });
+  };
+
+  handleCancel = () => {
+    this.setState({
+      visible: false,
+    });
+  };
+
+  showModal = () => {
+    this.setState({
+      visible: true,
+    });
+  };
+
+  onGenerate = () => {
+    const { configData } = this.props;
+    const { description } = this.state;
+    const stringProp = JSON.stringify(configData);
+    axios({
+      url: 'http://localhost:4466/',
```

Comment:
All endpoints for a query should be referenced from the `datastoreConfig` prop as part of the global redux store. We can define the endpoint within `mock/datastoreConfig.js` as an additional key called `backend`.

Reply:
@Anisha1234 This item still requires attention as we would like to reference the production endpoint for the GraphQL server via `datastoreConfig`